tst/testinstall/syntax/syntaxtree.tst

```diff
@@ -0,0 +1,10 @@
+# This test prints the names of the test functions that fail to produce the desired output
+gap> Read( Filename(DirectoriesLibrary( "tst/testinstall/syntax" ), "testsyntax.g" ) );
+gap> i := PositionsProperty([1..Length(trees)], i -> trees[i] <> expect_trees[i]);;
+gap> List(trees{ i }, x -> x.name);
+[  ]
```

Comment:
This fails, actually gives `[ "stat_test_if", "test_comobj" ]`

Reply:
The former failure is probably because we now optimize `if true` and `if false` away in certain cases.